### PR TITLE
Rename and separate queues/jobs into `PCS.WorkItems`

### PR DESCRIPTION
**Pull request by premun**
https://github.com/dotnet/arcade-services/issues/3807

- Renames jobs to work items to stop conflating them with ACA jobs
- Moves them out to a separate project so that we can possibly use queue processing in other projects (like dependency updater)

### CODE REVIEW

File: src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+{
+    private const string WorkItemQueueNameConfigurationKey = $"{WorkItemConsumerOptions.ConfigurationKey}:WorkItemQueueName";
+
+    public static void AddWorkItemQueues(this WebApplicationBuilder builder, DefaultAzureCredential credential, bool waitForInitialization)
+    {
+        builder.AddAzureQueueClient("queues", settings => settings.Credential = credential);
+
+        var queueName = builder.Configuration.GetRequiredValue(WorkItemQueueNameConfigurationKey);
+
+        // When running the service locally, the JobsProcessor should start in the Working state
```
Comment (oleksandr-didyk) [inline src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemConfiguration.cs:25]:
```suggestion
        // When running the service locally, the WorkItemProcessor should start in the Working state
```

File: src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemConsumer.cs [hunk 2]
```diff
@@ -82,13 +84,13 @@ private async Task ReadAndProcessJobAsync(QueueClient queueClient, JobScope jobS
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, "Processing job {jobId} attempt {attempt}/{maxAttempts} failed",
-                job.Id, message.DequeueCount, _options.Value.MaxJobRetries);
-            // Let the job retry a few times. If it fails a few times, delete it from the queue, it's a bad job
-            if (message.DequeueCount == _options.Value.MaxJobRetries)
+            _logger.LogError(ex, "Processing worki tem {workItemId} attempt {attempt}/{maxAttempts} failed",
```
Comment (oleksandr-didyk) [inline src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemConsumer.cs:87]:
```suggestion
            _logger.LogError(ex, "Processing work item {workItemId} attempt {attempt}/{maxAttempts} failed",
```

**Review by oleksandr-didyk — APPROVED** [review]

**Review by oleksandr-didyk — APPROVED** [review]

File: src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemProcessorState.cs [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 // The .NET Foundation licenses this file to you under the MIT license.
 
-namespace ProductConstructionService.Api.Queue;
+namespace ProductConstructionService.WorkItems;
 
-public enum JobsProcessorState
+public enum WorkItemProcessorState
 {
     /// <summary>
     /// The JobsProcessor is waiting for service to fully initialize
```
Comment (dkurepa) [inline src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemProcessorState.cs:9]:
```suggestion
    /// The WorkItemProcessor is waiting for service to fully initialize
```

File: test/ProductConstructionService.Api.Tests/WorkItemScopeTests.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+public class WorkItemScopeTests
+{
+    [Test]
+    public async Task WorkItemScopeRecordsMetricsTest()
+    {
+        IServiceCollection services = new ServiceCollection();
+
+        Mock<ITelemetryScope> telemetryScope = new();
+        Mock<ITelemetryRecorder> metricRecorderMock = new();
+        TestWorkItem textWorkItem = new() { Text = string.Empty };
```
Comment (dkurepa) [inline test/ProductConstructionService.Api.Tests/WorkItemScopeTests.cs:24]: nit
```suggestion
        TestWorkItem testWorkItem = new() { Text = string.Empty };
```

Comment (dkurepa) [reply test/ProductConstructionService.Api.Tests/WorkItemScopeTests.cs:24]: will be easier to do in vs tho

**Review by oleksandr-didyk — APPROVED** [review]

**Pull request merged**